File: MutableAI/JavaScript/Q3.js
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function(n) {
    //Reverse bits of a given 32 bits unsigned integer.
    
    let result = 0; 
    for (let i = 0; i < 32; i++) {
        result = result << 1;
        result = result | (n & 1);
        n = n >> 1;
    }
    return result;
}

File: MutableAI/Java/Q3.java
package Java;

public class Q3 {
    public int reverseBits(int n) {
         //Reverse bits of a given 32 bits unsigned integer.
         int result = 0;
         for (int i = 0; i < 32; i++) {
             result <<= 1;
             result |= n & 1;
             n >>= 1;
         }           
         return result;
    }
}
